smart_contracts/contracts/test/purse-holder-stored/src/main.rs

```diff
@@ -64,12 +66,23 @@ pub extern "C" fn call() {
         entry_points
     };
 
-    let (contract_hash, contract_version) = storage::new_contract(
-        entry_points,
-        None,
-        Some(HASH_KEY_NAME.to_string()),
-        Some(ACCESS_KEY_NAME.to_string()),
-    );
+    let (contract_hash, contract_version) = if !is_locked {
+        storage::new_contract(
+            entry_points,
+            None,
+            Some(HASH_KEY_NAME.to_string()),
+            Some(ACCESS_KEY_NAME.to_string()),
+        )
+    } else {
+        let contract_hash = storage::new_locked_contract(
+            entry_points,
+            None,
+            Some(HASH_KEY_NAME.to_string()),
+            Some(ACCESS_KEY_NAME.to_string()),
+        );
+        (contract_hash, CONTRACT_INITIAL_VERSION)
```

Comment:
Shouldn't be an issue to return the contract version inside `new_locked_contract` rather than ignoring it and returning a constant here?

Unless it was said otherwise I wouldn't mind if `create_contract_package_at_hash` (and `new_contract`) has `is_locked` argument to simplify this condition. For better readability it doesn't need to be a bool, but enum like: `enum ContractPackageStatus { Locked, Unlocked }` or something similar which at point of `ext_ffi` call could be converted into appropriate bool value. I.e.

```rust
fn create_contract_package_at_hash(status: ContractPackageStatus);
// ...
let (package_hash, access_key) = create_contract_package_at_hash(ContractPackageStatus::Locked);
```

Reply:
Changed from trivial `bool` to a better enum `ContractPackageStatus` for better readability